packages/vuelidate/src/core.js

```diff
@@ -217,58 +209,55 @@ function createValidatorResult (rule, state, key) {
  */
 function createValidationResults (rules, state, key, parentKey) {
   // collect the property keys
-  const ruleKeys = Object.keys(rules)
-
   const $dirty = ref(false)
 
-  let result = {
+  const results = {
     $dirty,
     $touch: () => { $dirty.value = true },
     $reset: () => { $dirty.value = false }
   }
 
-  /**
-   * If there are no validation rules, it is most likely
-   * a top level state, aka root
-   */
-  if (!ruleKeys.length) return result
+  watch(rules, (changedRules) => {
+    /**
+     * If there are no validation rules, it is most likely
+     * a top level state, aka root
+     */
+    const ruleKeys = Object.keys(changedRules)
+    // if (!ruleKeys.length) return result
+    ruleKeys.forEach(ruleKey => {
+      results[ruleKey] = createValidatorResult(
+        changedRules[ruleKey],
+        state,
+        key
+      )
+    })
 
-  ruleKeys.forEach(ruleKey => {
-    result[ruleKey] = createValidatorResult(
-      rules[ruleKey],
-      state,
-      key
+    results.$invalid = computed(() =>
+      ruleKeys.some(ruleKey => results[ruleKey].$invalid.value)
     )
-  })
-
-  result.$invalid = computed(() =>
-    ruleKeys.some(ruleKey => result[ruleKey].$invalid.value)
-  )
-
-  result.$pending = computed(() =>
-    ruleKeys.some(ruleKey => result[ruleKey].$pending.value)
-  )
-
-  result.$error = computed(() =>
-    result.$invalid.value && $dirty.value
-  )
-
-  result.$errors = computed(() => ruleKeys
-    .filter(ruleKey => result[ruleKey].$invalid.value)
-    .map(ruleKey => {
-      const res = result[ruleKey]
-      return reactive({
-        $propertyPath: parentKey ? `${parentKey}.${key}` : key,
-        $property: key,
-        $validator: ruleKey,
-        $message: res.$message,
-        $params: res.$params,
-        $pending: res.$pending
+    results.$pending = computed(() =>
+      ruleKeys.some(ruleKey => results[ruleKey].$pending.value)
+    )
+    results.$error = computed(() =>
+      results.$invalid.value && $dirty.value
+    )
+    results.$errors = computed(() => ruleKeys
+      .filter(ruleKey => results[ruleKey].$invalid.value)
+      .map(ruleKey => {
+        const res = results[ruleKey]
+        return reactive({
```

Comment:
Does this need to be `reactive`? It’s already inside a computed property, so I would assume it would stay updated. Unless you use that to unwrap the `res.$message` and others?

Reply:
It's is to expand the properties, correct. Otherwise they are with refs